src/care/ProfilePlugin.cpp

```diff
@@ -0,0 +1,63 @@
+#include "care/ProfilePlugin.h"
+
+#include "care/PluginData.h"
+
+/* CUDA profiling macros */
+#if defined(__CUDACC__) && CARE_HAVE_NVTOOLSEXT
+#include "nvToolsExt.h"
+#endif
+
+namespace care{
+   unsigned int ProfilePlugin::s_current_color = 0;
+   uint32_t ProfilePlugin::s_colors[7] = { 0x0000ff00, 0x000000ff, 0x00ffff00, 0x00ff00ff, 0x0000ffff, 0x00ff0000, 0x00ffffff };
+   int ProfilePlugin::s_num_colors = sizeof(s_colors) / sizeof(uint32_t);
+   bool ProfilePlugin::s_profile_host_loops = true; 
+   bool ProfilePlugin::registered = false;
+   
+   ProfilePlugin::ProfilePlugin() {}
+   
+   void ProfilePlugin::registerPlugin(bool reg) {
```

Comment:
The bool argument is misleading since a user might expect they could call this function with true to register the plugin, then later with false to deregister it.

```
void ProfilePlugin::registerPlugin() {
    static RAJA::util::PluginRegistry::add<care::ProfilePlugin> L ("Profile plugin", "Care plugin for profiling"); // This line will only get executed the first time this function is called (that is the nature of a static variable in a function)
}
```